affordable-housing/CSVToHTMLConverter.py

```diff
@@ -0,0 +1,148 @@
+import sys
+import csv
+import re
+import usaddress
+
+# @author: ashwin
+
+def stTypeToAbbrev(streetType):
+    if streetType == 'Avenue':
+        return 'AV'
+    elif streetType == 'Boulevard':
+        return 'BL'
+    elif streetType == 'Circle':
+        return 'CL'
+    elif streetType == 'Court':
+        return 'CT'
+    elif streetType == 'Drive':
+        return 'DR'
+    elif streetType == 'Expressway':
+        return 'EX'
+    elif streetType == 'Freeway':
+        return 'FY'
+    elif streetType == 'Highway':
+        return 'HY'
+    elif streetType == 'Lane':
+        return 'LN'
+    elif streetType == 'Loop':
+        return 'LP'
+    elif streetType == 'Place':
+        return 'PL'
+    elif streetType == 'Parkway':
+        return 'PY'
+    elif streetType == 'Road':
+        return 'RD'
+    elif streetType == 'Square':
+        return 'SQ'
+    elif streetType == 'Street':
+        return 'ST'
+    elif streetType == 'Terrace':
+        return 'TR'
+    elif streetType == 'Walkway':
+        return 'WW'
+    elif streetType == 'Way':
+        return 'WY'
+    else:
+        return streetType
+    
+
+def mapsUrlToHref(url, addressString):
+    retval = "<a href=\"" + url + " San Jose, CA\">" + addressString + "</a>"
+    return retval
+
+def javaScriptWriter():
+    retval = "<script>\r\nfunction loadDoc(streetName, street, streetType) {\r\n  var xhttp = new XMLHttpRequest();\r\n  \r\n\r\nvar settings = {\r\n  \"async\": true,\r\n  \"crossDomain\": true,\r\n  \"url\": \"http://www3.sanjoseca.gov/codeEnforcement/cets/New_ces_Results.asp\",\r\n  \"method\": \"POST\",\r\n  \"headers\": {\r\n    \"accept-language\": \"en-us\",\r\n    \"content-type\": \"application/x-www-form-urlencoded; Charset=UTF-8\",\r\n    \"Access-Control-Allow-Origin\" : \"http://localhost\"\r\n  },\r\n  \"data\": \"PropHouse=\" + streetName + \"&PropStreet=\" + street + \"&PropStreetType=\" + streetType\r\n}\r\n\r\n$.ajax(settings).done(function (response) {\r\n  console.log(response);\r\n\r\n});\r\n\r\n}\r\n</script>"
+    return retval
+    
+
+if len(sys.argv) < 3:
```

Comment:
All the code at the top level (not function definitions) should be wrapped in "if __name__ == "__main__:". Example:

http://www.diveintopython.net/getting_to_know_python/index.html#odbchelper.divein